contentgrid-appserver-application-model/src/main/java/com/contentgrid/appserver/application/model/Application.java

```diff
@@ -87,6 +93,14 @@ public class Application {
             if (relation instanceof ManyToManyRelation manyToManyRelation && !tables.add(manyToManyRelation.getJoinTable())) {
                 throw new DuplicateElementException("Duplicate table named %s".formatted(manyToManyRelation.getJoinTable()));
             }
+            if (relation.getSourceEndPoint().hasFlag(VisibleEndpointFlag.class) &&
+                    (relation instanceof OneToManyRelation || relation instanceof ManyToManyRelation)) {
+                getFilterForRelation(relation); // assert filter exists for redirects
```

Comment:
I made a separate ticket for Exception handling or the use of Optional
https://xenitsupport.jira.com/browse/ACC-2447